tests/test_namespace.py

```diff
@@ -0,0 +1,55 @@
+import os
+import pytest
+
+
+@pytest.mark.asyncio
```

Comment:
these should be parameterized tests as opposed to copy pastes of the same test class

Additionally, we should test with real tags and also tags with things like ids in the xpath, classnames, etc